[JIT] Allow del statements with multiple targets

SplitInfinity opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48876 [JIT] Allow del statements with multiple targets**

**Summary**
This commit adds support for `del` statements with multiple targets.
Targets are deleted left-to-right just like Python.

**Test Plan**
This commit updates the `TestBuiltins.test_del_multiple_operands` unit
test to actually test that multiple deletion works instead of asserting
that an error is thrown.

**Fixes**
This commit fixes #48635.

Differential Revision: [D25386285](https://our.internmc.facebook.com/intern/diff/D25386285)

Commit:
[JIT] Allow del statements with multiple targets

**Summary**
This commit adds support for `del` statements with multiple targets.
Targets are deleted left-to-right just like Python.

**Test Plan**
This commit updates the `TestBuiltins.test_del_multiple_operands` unit
test to actually test that multiple deletion works instead of asserting
that an error is thrown.

[ghstack-poisoned]

gmagogsfm commented on test/jit/test_builtins.py:
```diff
@@ -109,22 +109,31 @@ def fn(x):
+
+        py_out = del_list_multiple_operands([0, 1, 2])
+        jit_out = torch.jit.script(del_list_multiple_operands)([0, 1, 2])
+        self.assertEquals(py_out, jit_out)
```
Use `self.checkScript`?

SplitInfinity replied:
You can't - list inputs are mutable. A list input will be mutated by the script function and then reused by the eager mode function.

Commit:
Update on "[JIT] Allow del statements with multiple targets"


**Summary**
This commit adds support for `del` statements with multiple targets.
Targets are deleted left-to-right just like Python.

**Test Plan**
This commit updates the `TestBuiltins.test_del_multiple_operands` unit
test to actually test that multiple deletion works instead of asserting
that an error is thrown.

**Fixes**
This commit fixes #48635. 

[ghstack-poisoned]
